feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts

```diff
@@ -1,12 +1,43 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { BulkPricesService } from '../../core/services/bulk-prices.service';
+import { RoutingService } from '@spartacus/core';
+import { BulkPrice } from '../../core/model/bulk-price.model';
+import { Observable, Subscription } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-bulk-pricing-table',
   templateUrl: './bulk-pricing-table.component.html',
 })
-export class BulkPricingTableComponent implements OnInit {
-  constructor() {}
-  testString = 'bulk-pricing-table works!';
+export class BulkPricingTableComponent implements OnInit, OnDestroy {
+  pricingTiers: BulkPrice[];
+  dataStream: Subscription;
+  testString;
 
-  ngOnInit(): void {}
+  constructor(
+    private routingService: RoutingService,
+    private bulkPrices: BulkPricesService
+  ) {
+    this.testString = 'bulk-pricing-table works!';
+  }
+
+  ngOnInit(): void {
+    this.dataStream = this.getPrices().subscribe((p) => {
+      this.pricingTiers = p;
+      console.log({ p });
+    });
+  }
+
+  getPrices(): Observable<BulkPrice[]> {
+    return this.routingService.getRouterState().pipe(
```

Comment:
could we think of another way of optaining the product code without the usage of the router?

Reply:
Investigating, other options